bciers/apps/administration/app/components/contacts/ContactForm.tsx

```diff
@@ -82,25 +82,35 @@ export default function ContactForm({
             body: JSON.stringify(body),
           },
         );
+
         if (response.error) {
           setError(response.error);
-          // return error so SingleStepTaskList can re-enable the submit button and user can attempt to submit again
           return { error: response.error };
+        } else {
+          if (method === "POST") {
+            // Update formState with the new ID from the response
+            const updatedFormState = {
+              ...formState, // Retain the current form state
+              id: response.id, // add the id from the response
+            };
+            // Set the updated form state
+            setFormState(updatedFormState);
+          }
```

Comment:
Can we remove this and just move the `updatedFormState` code to the `if (isCreatingState)` block below?